lib/client-assets.php

```diff
@@ -929,7 +929,7 @@ function gutenberg_editor_scripts_and_styles( $hook ) {
 	// Initialize the editor.
 	$gutenberg_theme_support = get_theme_support( 'gutenberg' );
 	$align_wide              = get_theme_support( 'align-wide' );
-	$color_palette           = get_theme_support( 'editor-color-palette' );
+	$color_palette           = current( (array) get_theme_support( 'editor-color-palette' ) );
```

Comment:
I think this may be breaking back compatibility with the existing version:
``` 
add_theme_support( 'editor-color-palette',
 	'#fff'
 	'#f00'
);
```
Maybe we should handle some condition to be backcompatible.

Reply:
My question would be, is this required? Gutenberg is still in development and from what I can see the theme support declaration have already changed. (Though, I don't know if it was made backwards compatible.)

Also, is this backwards compatible too?:
```
add_theme_support( 'editor-color-palette',
	array(
		'name' => 'color_name_1',
		'color' => '#color1',
	),
	array(
		'name' => 'color_name_2',
		'color' => '#color2',
	)
);
```

If needed, I can surely, edit my code to incorporate backwards compatibility.

Reply:
Hi all,
Thank you for sharing your thoughts, and for this contributions.
When changing colors mechanism, we always made efforts to try to be back-compatible in this case being back-compatible does not look very complicated. I think If we add this block:
```php
	// Backcompat for Color Palette set as multiple parameters.
	if ( is_string( $color_palette ) || isset( $color_palette['color'] ) ) {
		$color_palette = get_theme_support( 'editor-color-palette' );
		wp_add_inline_script(
			'wp-edit-post',
			'console.warn( "' .
				__( 'Setting colors using multiple parameters is deprecated. Please pass a single parameter with an array of colors. See https://wordpress.org/gutenberg/handbook/extensibility/theme-support/ for details.', 'gutenberg' ) .
			'");'
		);
	}
```
Bellow `$color_palette           = current( (array) get_theme_support( 'editor-color-palette' ) );``, it should work with both our old versions:
```
add_theme_support( 'editor-color-palette',
	array(
		'name => 'red'
		'color' => '#ff0',
	),
	array(
		'name => 'green'
		'color' => '#0f0',
	),
	array(
		'name => 'blue'
		'color' => '#00f',
	)
);
```
and
```
add_theme_support( 'editor-color-palette',
	'#f00',
	'#0f0',
	'#00f'
);
```

Would it be possible to rebase this branch so we can test with our latest changes, e.g., the mechanism to hide color palettes if we don't have colors to choose?

@mtias what are your thoughts? In your opinion, should we advance with this changes and set the color palette using a single array parameter instead of the current way (multiple parameters)?